### PR TITLE
Add FarmlandHydrationChangeEvent

### PR DESCRIPTION
**Pull request by DebHugo**
## Introduction
Add suggestion of #5883 

### API changes
Add FarmlandHydrationChangeEvent class in `pocketmine\event\block` and modif `onRandomTick` function in Farmland class

### CODE REVIEW

**Comment by dktapps** [comment]
`FarmlandHydrationChangeEvent` would be better. `wetness` sounds weird.

**Review by ShockedPlot7560 — DISMISSED** [review]

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	public function __construct(
+		Block $block,
+		private int $oldWetness,
+		private int $newWetness,
+	){
+		parent::__construct($block);
+	}
+
+	public function getOldWetness() : int{
```
Comment (dktapps) [inline src/event/block/FarmlandHydrationChangeEvent.php]: This should be named `getOldHydration()` to be consistent with the event name

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		return $this->oldWetness;
+	}
+
+	public function getNewWetness() : int{
```
Comment (dktapps) [inline src/event/block/FarmlandHydrationChangeEvent.php]: Same again here

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+use pocketmine\block\Farmland;
+
+/**
+ * Called when farmland wetness is updated.
+ */
+class FarmlandHydrationChangeEvent extends BlockUpdateEvent{
+
+	public function __construct(
+		Block $block,
+		private int $oldWetness,
```
Comment (ShockedPlot7560) [inline src/event/block/FarmlandHydrationChangeEvent.php]: Could be better to rename it `$oldHydration` ?

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		private int $newWetness,
```
Comment (ShockedPlot7560) [inline src/event/block/FarmlandHydrationChangeEvent.php]: Same here

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+declare(strict_types=1);
+
+namespace pocketmine\event\block;
+
+use pocketmine\block\Block;
+use pocketmine\block\Farmland;
+
+/**
+ * Called when farmland wetness is updated.
```
Comment (ShockedPlot7560) [inline src/event/block/FarmlandHydrationChangeEvent.php]:
```suggestion
 * Called when farmland hydration is updated.
```

File: src/event/block/FarmlandHydrationChangeEvent.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace pocketmine\event\block;
+
+use pocketmine\block\Block;
+use pocketmine\block\Farmland;
+
+/**
+ * Called when farmland hydration is updated.
+ */
+class FarmlandHydrationChangeEvent extends BlockUpdateEvent{
```
Comment (dktapps) [inline src/event/block/FarmlandHydrationChangeEvent.php]: Why is this extending BlockUpdateEvent?

Comment (DebHugo) [reply src/event/block/FarmlandHydrationChangeEvent.php]: Because I'm making a mistake. I change this now to extend `BlockEvent`

**Pull request merged**